tests/worker/test_workflow.py

```diff
@@ -5088,3 +5089,176 @@ async def test_workflow_current_update(client: Client, env: WorkflowEnvironment)
         assert {"update1", "update2", "update3", "update4", "update5"} == set(
             await handle.result()
         )
+
+
+@workflow.defn
+class UnfinishedHandlersWorkflow:
+    def __init__(self):
+        self.started_handler = False
+        self.handler_may_return = False
+        self.handler_finished = False
+
+    @workflow.run
+    async def run(self, wait_for_handlers: bool) -> bool:
+        await workflow.wait_condition(lambda: self.started_handler)
+        if wait_for_handlers:
+            self.handler_may_return = True
+            await workflow.wait_condition(workflow.all_handlers_finished)
+        return self.handler_finished
+
+    async def _do_update_or_signal(self) -> None:
+        self.started_handler = True
+        await workflow.wait_condition(lambda: self.handler_may_return)
+        self.handler_finished = True
+
+    @workflow.update
+    async def my_update(self) -> None:
+        await self._do_update_or_signal()
+
+    @workflow.update(unfinished_policy=workflow.HandlerUnfinishedPolicy.ABANDON)
+    async def my_update_ABANDON(self) -> None:
+        await self._do_update_or_signal()
+
+    @workflow.update(
+        unfinished_policy=workflow.HandlerUnfinishedPolicy.WARN_AND_ABANDON
+    )
+    async def my_update_WARN_AND_ABANDON(self) -> None:
+        await self._do_update_or_signal()
+
+    @workflow.signal
+    async def my_signal(self):
+        await self._do_update_or_signal()
+
+    @workflow.signal(unfinished_policy=workflow.HandlerUnfinishedPolicy.ABANDON)
+    async def my_signal_ABANDON(self):
+        await self._do_update_or_signal()
+
+    @workflow.signal(
+        unfinished_policy=workflow.HandlerUnfinishedPolicy.WARN_AND_ABANDON
+    )
+    async def my_signal_WARN_AND_ABANDON(self):
+        await self._do_update_or_signal()
+
+
+async def test_unfinished_update_handlers(client: Client):
+    async with new_worker(client, UnfinishedHandlersWorkflow) as worker:
+        await _UnfinishedHandlersTest(client, worker, "update").run_test()
+
+
+async def test_unfinished_signal_handlers(client: Client):
+    async with new_worker(client, UnfinishedHandlersWorkflow) as worker:
```

Comment:
```suggestion
    if env.supports_time_skipping:
        pytest.skip(
            "Java test server: https://github.com/temporalio/sdk-java/issues/1903"
        )
    async with new_worker(client, UnfinishedHandlersWorkflow) as worker:
```

Java test server not very update capable currently